grails-app/controllers/org/pih/warehouse/JsonController.groovy

```diff
@@ -979,7 +979,7 @@ class JsonController {
 				def productQuantity = 0
 				// We need to check to make sure this is a valid product
 				def inventoryItemList = []
-				if (product.id) {
+				if (product.id && params.skipQuantity != "true") {
```

Comment:
The logic here seems backwards. Also, you should also not be doing string comparison here. Use `params.boolean("skipQuantity")` instead. 
```
boolean skipQuantity = params.boolean("skipQuantity")?:false
if (product.id && !skipQuantity) { 
...
}
```

Looking at the rest of the code, it doesn't look like we're even calculating quantity anywhere. So perhaps something else is really slow. I believe the code-join on InventoryItem is definitely a problem (so keeping the "skip" logic is probably good) but perhaps we need to look at the productSearch a little more closely as well.

Reply:
http://mrhaki.blogspot.com/2010/02/grails-goodness-type-conversion-on.html